src/main/webapp/app/course/manage/course-storage.service.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { Course } from 'app/entities/course.model';
+import { SubjectObservablePair } from 'app/utils/rxjs.utils';
+
+@Injectable({ providedIn: 'root' })
+export class CourseStorageService {
+    private storedCourses: Course[] = [];
+
+    private readonly courseUpdateSubscriptions: Map<number, SubjectObservablePair<Course>> = new Map();
+
+    setCourses(courses: Course[] | null) {
```

Comment:
we prefer to use `undefined` over `null`
please replace all occurrences of `null` with `undefined` in ts / html files in the whole PR
(it should be possible to use `courses?: Course[]` then)